## Задание 1

В рамках комплексного анализа игры некоторого футбольного клуба, тренерский штаб решил исследовать зависимость между весом игроков (в килограммах) и дистанцией (в метрах), которую игроки пробегают за первый тайм. Данные приведены в таблице. Проверить гипотезу о независимости веса игроков и преодоленной дистанции, считая, что данные имеют нормальное распределение, против альтернативы о наличии отрицательной зависимости. Примите уровень значимости равным 0,05.

In [1]:
import pandas as pd
import numpy as np
from scipy import stats

In [2]:
data1 = pd.read_excel('data.xlsx')
Xy = [i.values for i in data1['Weight'],data1['Distanse']]
data1

Weight  Distanse
0     67.3      8828
1     73.1      8332
2     72.6      7101
3     73.9      8127
4     74.7      5985
5     75.0     10827
6     77.1      8565
7     90.4      4295
8     70.9      4750
9     74.7      6379
10    80.2     10103

In [3]:
stats.pearsonr(Xy[0], Xy[1])

(-0.27007983715504258, 0.42185710831232665)

Коэффицент корреляции Пирсона $r_{x,y}=\frac{\sum_{i=1}^{m}(x_{i}-\bar{x})  (y_{i}-\bar{y})}{\sqrt{\sum_{i=1}^{m}(x_{i}-\bar{x})^2  (y_{i}-\bar{y})^2}}=\frac{cov(x,y)}{\sqrt{S_{x}S_{y}}}$

In [4]:
def korP(V):
    cov= sum([(V[0][i] - np.mean(V[0]))*(V[1][i] - np.mean(V[1])) for i in range(len(V[0]))])
    print u'Ковариация cov(x,y) =',cov
    Sx=sum([(V[0][i] - np.mean(V[0]))**2 for i in range(len(V[0]))])
    Sy=sum([(V[1][i] - np.mean(V[1]))**2 for i in range(len(V[0]))])
    print u'\nВыборочные дисперсии Sx= ',Sx,u'\nСредневыборочное Sy = ',Sy
    r=cov/np.sqrt(Sx*Sy)
    print u'Коэффициент корреляции Пирсона r =', r
    return r
r=korP(Xy)

Ковариация cov(x,y) = -33410.4

Выборочные дисперсии Sx=  353.287272727 
Средневыборочное Sy =  43316268.0
Коэффициент корреляции Пирсона r = -0.270079837155


Связи между признаками могут быть слабыми и сильными (тесными). Их критерии оцениваются по шкале Чеддока:

0.1 < $r_{xy}$ < 0.3: слабая;

0.3 < $r_{xy}$ < 0.5: умеренная;

0.5 < $r_{xy}$ < 0.7: заметная;

0.7 < $r_{xy}$ < 0.9: высокая;

0.9 < $r_{xy}$ < 1: весьма высокая;

В нашем задании связь между дистанцией и весом слабая и обратная.

$H_{0}: r_{xy}$ = 0, нет линейной взаимосвязи между переменными;

$H_{1}: r_{xy}$ ≠ 0, есть линейная взаимосвязь между переменными;

In [5]:
t=np.abs(r)*(np.sqrt(len(Xy[0])-2)/np.sqrt(1-r**2))
print t

0.841511832153



$t_{spec}$ = 0.841511832153

По таблице Стьюдента с уровнем значимости α=0.05 и степенями свободы k=9 находим $t_{crit}$ :

$t_{crit}$ (n-m-1;α/2) = (9;0.025) = 2.262

Поскольку |$t_{spec}$| < $t_{crit}$, то принимаем гипотезу о равенстве 0 коэффициента корреляции. Другими словами, коэффициент корреляции статистически - не значим

## Задание 2

2.	Имеются данные о баллах за контрольную работу 1 (СВ X) и об итоговых баллах (СВ Y) некоторой группы студентов бизнес-информатики по Анализу данных. Проверить гипотезу о независимости СВ X и Y на уровне значимости 0,05.

In [6]:
data2 = pd.read_excel('data2.xlsx')
Xy2 = [i.values for i in data2['Test1'],data2['Sumup']]
data2

Test1  Sumup
0       5     29
1      28     89
2       8     48
3      28     95
4       5     38
5      27     86
6      30     75
7      29     86
8      25     86
9       4     51
10     28     75

#### Коэффициент Кендэла 

Ставим ранги

In [7]:
data2=data2.sort_values(by='Test1')
data2['RankX']=[i+1 for i in range(len(Xy2[0]))]
data2=data2.sort_values(by='Sumup')
data2['Ranky']=[i+1 for i in range(len(Xy2[0]))]
data2=data2.sort_values(by='Test1')

Вручную посчитал Concordant pairs и Disconcordant pairs

In [8]:
data2['C']=[7,9,8,7,4,3,1,0,2,0,0]
data2['D']=[3,0,0,0,2,2,3,3,0,1,0]

In [9]:
data2

Test1  Sumup  RankX  Ranky  C  D
9       4     51      1      4  7  3
0       5     29      2      1  9  0
4       5     38      3      2  8  0
2       8     48      4      3  7  0
8      25     86      5      7  4  2
5      27     86      6      8  3  2
10     28     75      9      5  1  3
1      28     89      7     10  0  3
3      28     95      8     11  2  0
7      29     86     10      9  0  1
6      30     75     11      6  0  0

$\gamma=\frac{P-Q}{\frac{1}{2}N(N-1)}$

In [10]:
P=sum(data2['C'].values)
Q=sum(data2['D'].values)
g=(P-Q)/(0.5*len(data2)*(len(data2)-1))
g

0.49090909090909091

Проверим

$H_{0}: \gamma$ = 0, нет линейной взаимосвязи между переменными;

$H_{1}: \gamma$ ≠ 0, есть линейная взаимосвязь между переменными;

$T_{cr} = z_{cr}\sqrt{\frac{2(2n+5)}{9n(n-1)}}$ 

$\Phi(z_{cr}) = (1-\alpha)/2 = 0.475$

По таблице Лапласа $ z_{cr} = 1.96$ =>

$T_{cr} = 0.46$

Т.к $\gamma > T_{cr}$ => отвергаем $H_{0}$ , корреляционная связь между оценками по двум тестам значимая

## Задание 3

In [11]:
data3 = pd.read_excel('data3.xlsx')
Xy3 = [i.values for i in data3['CREA'],data3['TG']]
p,pr=stats.spearmanr(Xy3[0],Xy3[1])

In [12]:
data3.head(5)

CREA    TG
0    21  1.29
1    21  1.14
2    12  1.05
3    18  1.05
4    12  0.84

In [13]:
p

0.30029461361660664

Связи между признаками могут быть слабыми и сильными (тесными). Их критерии оцениваются по шкале Чеддока:

0.1 < $p_{xy}$ < 0.3: слабая;

0.3 < $p_{xy}$ < 0.5: умеренная;

0.5 < $p_{xy}$ < 0.7: заметная;

0.7 < $p_{xy}$ < 0.9: высокая;

0.9 < $p_{xy}$ < 1: весьма высокая;

Связь между признаком Y и фактором X умеренная и прямая

$H_{0}: P_{xy}$ = 0, нет линейной взаимосвязи между переменными;

$H_{1}: p_{xy}$ ≠ 0, есть линейная взаимосвязь между переменными;

$T = t(\alpha,k)\sqrt{\frac{1-p^2}{n-2}}$, k = n-2 = 58

$t(α/2, k) = (0.05/2;58) = 2$


$T = 0.25$

Поскольку T < p, то отклоняем гипотезу $H_{0}$, корреляционная связь между оценками по двум тестам значимая